ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_layout/runs/$runIndex/index.tsx

```diff
@@ -97,7 +109,34 @@ const RunComponent = () => {
     return;
   }
 
-  const vulnTotal = vulnerabilities?.pagination.totalCount;
+  const vulnTotal = vulnerabilities.pagination.totalCount;
+
+  if (packagesIsPending) {
+    return <LoadingIndicator />;
+  }
+
+  if (packagesIsError) {
+    toast.error('Unable to load data', {
+      description: <ToastError error={packagesError} />,
+      duration: Infinity,
+      cancel: {
+        label: 'Dismiss',
+        onClick: () => {},
+      },
+    });
+    return;
+  }
+
+  const packagesTotal = packages.pagination.totalCount;
+
+  // Find deduplicated package types in packages data.
+  const detectedPackageManagers = [
+    ...new Set(
+      packages.data.map((item) => item.identifier?.type).filter(Boolean)
+    ),
+  ];
```

Comment:
This likely doesn't work as the query is paginated, so `packages` only contains the first page with the default size.

Reply:
Should this rather be changed to describe that the list is of the package sources, or something else that would describe it better? I believe to type to just be [ORT Identifier's type](https://github.com/oss-review-toolkit/ort/blob/a03471c7c5981bb1950d76f708bd151ca196dfbd/model/src/main/kotlin/Identifier.kt#L32-L37), so data to do what you're suggesting is not available here. The same is probably true for NPM/Yarn/pnpm all using NPM packages.

Reply:
> I believe to type to just be [ORT Identifier's type](https://github.com/oss-review-toolkit/ort/blob/a03471c7c5981bb1950d76f708bd151ca196dfbd/model/src/main/kotlin/Identifier.kt#L32-L37), so data to do what you're suggesting is not available here.

The information about whether an `Identifier` belongs to a package or project is not available in the `Identifier` itself. What the code *should* do is to collect the types of the identifiers of all root projects in an ORT result instead of iterating over packages.

Reply:
> > I believe to type to just be [ORT Identifier's type](https://github.com/oss-review-toolkit/ort/blob/a03471c7c5981bb1950d76f708bd151ca196dfbd/model/src/main/kotlin/Identifier.kt#L32-L37), so data to do what you're suggesting is not available here.
> 
> The information about whether an `Identifier` belongs to a package or project is not available in the `Identifier` itself. What the code _should_ do is to collect the types of the identifiers of all root projects in an ORT result instead of iterating over packages.

IMO there can be value for both, showing what package managers are used to manage the project and what type of dependencies the project uses.

Reply:
So maybe for this part it would make sense to reword the text to something like "100 packages from 2 _registries_ (Npm, Maven)". The package managers used by the project can be added here or somewhere else when the data is available, but I wouldn't hold this PR to add that.

Reply:
> IMO there can be value for both

Maybe, but for practical matters I've always found it to be much more useful to know which package managers / project types are used instead of which ecosystems packages stem from.

> So maybe for this part it would make sense to reword the text

I agree that this initial implementation could "cheat" by adjusting the text to the current implementation instead of making the implementation match my more complex ask. However, I'd propose a different wording, and say "100 packages of 2 types (Maven, NPM) found". Oh, and the list of package types should be sorted alphabetically to be deterministic.

Reply:
> Oh, and the list of package types should be sorted alphabetically to be deterministic.

Sorted by package count per type could be good and also deterministic.

Reply:
Agreed, but I'd also be fine with not having per-type counts in this first iteration. At least not in the card; that's probably too detailed information that could go to a header on the "Packages" tab.

Reply:
> "100 packages of 2 types (Maven, NPM) found"

How about "100 packages from 2 ecosystems (Maven, NPM) found" instead?

Reply:
I was at first thinking about almost exactly the same as I was using "ecosystems" myself [above](https://github.com/eclipse-apoapsis/ort-server/pull/1125#discussion_r1781198645), but then I didn't want to coin yet another new term. On the other hand, we should have the liberty to use "new" terms in the UI if we believe the (non-ORT) user is familiar with them.

TL;DR  "100 packages from 2 ecosystems (Maven, NPM)" would be ok with me.